### ISSUE TITLE
compile issue on VS 17.2

### ISSUE THREAD

**Issue by Zangetsu38** [merryhime/dynarmic#678]
# About:
after update VS on 17.2 get More 100 error on compile dynarmic

# Log
[message(6).txt](https://github.com/merryhime/dynarmic/files/8672952/message.6.txt)

**Comment by merryhime** [comment]
MSVC compiler bug regarding visibility of `using` within lambda scope.

**Comment by merryhime** [comment]
Latest version of the compiler I have is 19.32.31302 which does not seem to have this issue.

**Comment by Zangetsu38** [comment]
no idea, for me is just compiler is more susceptible and no accepte using value inside lambda :man_shrugging:

**Comment by merryhime** [comment]
Minimal test case: <https://godbolt.org/z/6exaqMhr3>

**Comment by merryhime** [comment]
Reported upstream: https://developercommunity.visualstudio.com/t/Type-alias-lookup-failure-within-paramet/10039150

**Comment by Zangetsu38** [comment]
master of ,vita3K now rip i imagine all case for all githubaction :D
![image](https://user-images.githubusercontent.com/5261759/168077302-af60643d-776a-4cfc-b1be-e77ad710b63a.png)

**Comment by Zangetsu38** [comment]
oh, Microsoft is speed lol
![image](https://user-images.githubusercontent.com/5261759/168082687-57a70017-94b1-4638-bde8-972d6917f831.png)

**Comment by merryhime** [comment]
Workaround in b224fad17136d1e0933f4ad23da1eb7a12db3e5b. Will close only when this can be reverted.

**Comment by Zangetsu38** [comment]
lol i am very shit, my workaround is so complicated
https://github.com/Vita3K/dynarmic/commit/66089b60cb832194d6c2ad289f164548587b9836

**Comment by vitor-k** [comment]
Seems to be fixed on Visual Studio 17.3, from my testing.
I checked the issue page that said it was fixed, so I reverted the workaround and tested before and after the update, resulting in (predictably) erroring before updating, and compiling fine after the update.
For the compiler, cmake reports `The CXX compiler identification is MSVC 19.33.31629.0`.
It also looks like github actions hasn't been updated yet, as it still errors out and reports `MSVC 19.32.31333.0` (Tested with Citra's CI), but hopefully it will be soon.
